### PR TITLE
Fix for Pulsar-JMS library producer.send() method

### PR DESCRIPTION
**Pull request by mukesh-ctds**
Fixes: [LS-1292](https://datastax.jira.com/browse/LS-1292)

[LS-1292]: https://datastax.jira.com/browse/LS-1292?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by eolivelli — CHANGES_REQUESTED** [review]
I am pretty sure that this change will break some tests.

I tihnk taht the current behavior is correct.

Can you please provide more context and the reason for such modifications ?

**Comment by mukesh-ctds** [comment]
I reviewed the JMS specification, and according to it, the JMSCorrelationID, JMSReplyTo, and JMSType properties should be set by the client rather than the `send` method. However, the current implementation sets these values at the producer level, not at the message level. By default, these properties on the producer object are null, so they are being set to null regardless of whether the message has values for these properties.

In short, under the current behavior, these properties are configured at the producer level, and there is no way to set them at the message level even if desired. The customer expects these values to be set at the message level.

File: pulsar-jms/src/main/java/com/datastax/oss/pulsar/jms/PulsarJMSProducer.java [hunk 1]
```diff
@@ -192,9 +192,6 @@ private void getProducerAndSend(Destination destination, Message message) throws
       message.setObjectProperty(prop.getKey(), prop.getValue());
     }
     message.setJMSPriority(priority);
-    message.setJMSCorrelationIDAsBytes(correlationID);
```
Comment (eolivelli) [inline pulsar-jms/src/main/java/com/datastax/oss/pulsar/jms/PulsarJMSProducer.java:195]: https://jakarta.ee/specifications/messaging/3.0/apidocs/jakarta/jms/jmsproducer

> message headers set using these methods will override any message headers that have been set directly on the message

this seems to imply that it is possible to set the correlationid on the message

According to the specs I would code it this way:

if (!message.hasCorrelationId()){
 message.setJMSCorrelationIDAsBytes(correlationID);
}

**Review by eolivelli — CHANGES_REQUESTED** [review]

**Review by eolivelli — APPROVED** [review]
LGTM
